### PR TITLE
Emic 261

### PR DESCRIPTION
**Pull request by ajstanley**
Non destructive changes need by EMiCdora.
Solr results will not return a fragment on a wild card search,
The delete hook no longer wipes out the entire #submit array, it merely removes the default.

### CODE REVIEW

File: islandora_book.module [hunk 1]
```diff
@@ -269,12 +269,13 @@ function islandora_book_islandora_bookCModel_islandora_solr_object_result_alter(
 
   if ($query_processor->solrQuery != " " && $query_processor->solrQuery && !$field_search) {
     $ia_view = variable_get('islandora_internet_archive_bookreader_default_page_view', "1");
-    $search_results['object_url_fragment'] = "page/1/mode/{$view_types[$ia_view]}/search/";
-    $search_results['object_url_fragment'] .= rawurlencode($query_processor->solrQuery);
+    if ($query_processor->solrQuery != '*:*') {
+      $search_results['object_url_fragment'] = "page/1/mode/{$view_types[$ia_view]}/search/";
+      $search_results['object_url_fragment'] .= rawurlencode('Two');
```
Comment (jordandukart) [inline islandora_book.module]: Err this seems very destructive as compared to the former.

Comment (ajstanley) [inline islandora_book.module]: The former, by setting the entire array to the new value would remove the current array, which contains (by default) islandora_delete_object_form_submit.  The replacement method explicitly removes that function, but leaves any others that might be there intact.

Comment (jordandukart) [inline islandora_book.module]: The search results `rawurlencode('Two')` change is what I was commenting on.

Comment (ajstanley) [inline islandora_book.module]: Debug line left in.  My bad.

**Pull request merged**